### ISSUE TITLE
How to handle click inside the menubar window?

### ISSUE THREAD

**Issue by romaincointepas** [pronebird/NSEventMonitor#3]
Right now with your example code, clicking inside the menubar window will also triggers the callback of `.start`.

A solution that could work would be to add a position object (x, y) representing the click coordinates on the screen to the `.start` callback so when the callback is called, we can check if the click was inside the window or not and act accordingly.

Would it also make sense to allow the `.start` callback to be fired more than once until we manually stop it? Right now it seems to only fire once (that would allow ignoring clicks inside the app menubar window while still listening for future clicks that may or may not be inside that menubar window).

**Comment by pronebird** [comment]
You should not be receiving clicks related to your menubar window. You can test that in example app by putting `console.log` inside of event handler passed to `.start`.

I can only see that happen if you created `BrowserWindow` from renderer process but installed monitoring in main process or vise-versa. For that scenario I agree that proxying NSEvent's fields would be beneficial.

Also, when you create new `NSEventMonitor`, make sure you store it somewhere for as long as you need monitoring, i.e. store it in global variable or within relevant Object.

Once `.start`ed `NSEventMonitor` will fire events until you explicitly call `.stop`.

**Comment by romaincointepas** [comment]
Thanks for answering. I was using [maxogden/menubar](https://github.com/maxogden/menubar) and had quite a setup (both a menubar window on click, and a context menu on right-click). I could sometimes reproduce when stress-testing (doing many clicks fast, but at no point I was calling `.stop()`, and the events would suddenly stop coming). Not sure what was the problem, I refactored and I can't seem to reproduce anymore, so it was probably something in my code.

Closing this, sorry about that!
